Client/Frontend/Browser/TextContentDetector.swift

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+class TextContentDetector {
+
+    enum DetectedType {
+        case phoneNumber(String)
+        case link(URL)
+    }
+
+    private static let dataDetector: NSDataDetector = {
+        let types: NSTextCheckingResult.CheckingType = [.link, .phoneNumber]
+        return try! NSDataDetector(types: types.rawValue)
+    }()
+
+    static func detectTextContent(_ content: String) -> DetectedType? {
+
+        guard let match = dataDetector.firstMatch(in: content, options: [], range: NSRange(location: 0, length: content.count)) else { return nil }
+        switch match.resultType {
+        case .link:
+            return .link(match.url!)
```

Comment:
I am surprised these are optional,. I think the force-unwrap is ok because it doesn't make sense for the resultType to not have a filled url/phonenum

Reply:
I respected the code presented in the `NSDataDetector` documentation page. I guess this is on old class rarely used that does not offer a proper safe swift wrapping yet.